ml_metadata/tools/mlmd_bench/fill_types_workload_test.cc

```diff
@@ -23,59 +25,431 @@ limitations under the License.
+// the specified number of operations.
+TEST_P(FillTypesUpdateParameterizedTestFixture, SetUpImplWhenDbIsEmptyTest) {
+  TF_ASSERT_OK(fill_types_update_->SetUp(store_.get()));
+  // Gets the number of types inside db after SetUpImpl().
+  std::vector<Type> existing_types;
+  TF_ASSERT_OK(GetExistingTypes(GetParam().fill_types_config(), store_.get(),
+                                existing_types));
+  EXPECT_EQ(existing_types.size(), GetParam().num_operations());
+  EXPECT_EQ(GetParam().num_operations(), fill_types_update_->num_operations());
+}
+
+// Tests the RunOpImpl() for FillTypes update cases when db is empty in the
+// first place.
+// Checks indeed all the work items have been executed and there are certain
+// number of existed types inside db have been updated.
+TEST_P(FillTypesUpdateParameterizedTestFixture, UpdateWhenDbIsEmptyTest) {
+  TF_ASSERT_OK(fill_types_update_->SetUp(store_.get()));
+  // Gets the existing types in db after set up but before update for later
+  // comparison. The `existing_types_before_update` will contains the types that
+  // were made up and inserted in the SetUpImpl().
+  std::vector<Type> existing_types_before_update;
+  TF_ASSERT_OK(GetExistingTypes(GetParam().fill_types_config(), store_.get(),
+                                existing_types_before_update));
+
+  // Executes the update workloads.
+  TF_ASSERT_OK(ExecuteWorkload(store_.get(), fill_types_update_));
+
+  // Gets the existing types in db after update for later comparison.
+  std::vector<Type> existing_types_after_update;
+  TF_ASSERT_OK(GetExistingTypes(GetParam().fill_types_config(), store_.get(),
+                                existing_types_after_update));
+
+  // The update should update all the made up and inserted types in the
+  // SetUpImpl() and not introduce any new types in the db.
+  ASSERT_EQ(existing_types_before_update.size(),
+            existing_types_after_update.size());
+
+  // For all the types inside db, they should be updated. Checks their update
+  // status.
+  for (int64 i = 0; i < (int64)existing_types_after_update.size(); ++i) {
+    switch (GetParam().fill_types_config().specification()) {
+      case FillTypesConfig::ARTIFACT_TYPE: {
+        EXPECT_TRUE(CheckTypesUpdateStatus<ArtifactType>(
+            existing_types_before_update[i], existing_types_after_update[i]));
+        break;
+      }
+      case FillTypesConfig::EXECUTION_TYPE: {
+        EXPECT_TRUE(CheckTypesUpdateStatus<ExecutionType>(
+            existing_types_before_update[i], existing_types_after_update[i]));
+        break;
+      }
+      case FillTypesConfig::CONTEXT_TYPE: {
+        EXPECT_TRUE(CheckTypesUpdateStatus<ContextType>(
+            existing_types_before_update[i], existing_types_after_update[i]));
+        break;
+      }
+      default:
+        LOG(FATAL) << "Wrong specification for FillTypes configuration input "
+                      "in testing !";
+    }
+  }
```

Comment:
move these test check into a function, so it can be shared with the rest of three update tests. wdyt